Assignment5Classes/src/retailclasses/Store.java

```diff
@@ -0,0 +1,90 @@
+
+package retailclasses;
+
+import java.util.Scanner;
+
+/**
+ * Program to store item information for RetailItem
+ * @author Seth Frosch
+ */
+public class Store {
+    
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String[] args) {
+        /**
+         * @param description A string field for a description of the item
+         * @param unitsOnHand An integer for the amount of an item in the 
+         * inventory
+         * @param price A double for the price of the item
+         * @param keyboard Will accept user input
+         */
+        String description;
+        int unitsOnHand;
+        double price;
+        Scanner keyboard = new Scanner(System.in);
+        
+        /**
+         * Outputs prompts for user to enter description, amount on hand, and
+         * price for item 1
+         */
+        System.out.println("Enter the description for item 1:");
+        description = keyboard.nextLine();
```

Comment:
@srtinkey There are two reasons to create variables as close to usage as possible:

1. In the event of a large driver(main file) it makes the visible read distance between the declaration and usage of a variable much smaller(don't have to scroll). I am sure at some point you may have been taught to put all your declarations at the top of your file. This is an old behavior. They should be listed at the top of your usage block.
2. To make it easier to extract logic into methods. If we look at each block of assignment and scanner reading they are effectively the same. The DRY rule says "Don't repeat yourself" these blocks would be considered repetitive and a single method definition could encapsulate this logic. It could take scanner as an argument and perform the read and assign variables finally returning a new `RetailItem` instance. In each method body as a result would create its own variables for `description`, `unitsOnHand`, and `price` on each invocation.

If each of the declarations was near the logic group you would see that it is very repetitive and that would directly influence you to extract a method.

Java has garbage collection(GC) for variables as well. This means it returns allocated memory(variable space) to the computer after a variable falls out of scope. In other languages like C++ GC is something new and isn't always taught so using the same variable over and over keeps you from wasting heap space(memory) because you forgot to deallocate(manually give the memory back) the memory.

GC runs on an interval but it is easier for the JVM to know what is out of scope when the logic is encapsulated in a method. This means that after the method is invoked 3 times the original variable space used by `description`, `unitsOnHand`, and `price` will be returned to the system and all that will remain is the storage used for the `RetailItem` object.

Especially since primitives are pass by value we get a new copy in memory for the Object instance as well as what is defined in the driver.

This rule is a little less effective when your object has fields to a bunch of other objects but the GC still works the same.

TLDR;
You should consider putting all of the logic in your driver in methods and the bulk of your driver will be control statements(if, loops) and method invocations.